### PR TITLE
[Intl] Improve the error message when country code is wrong

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #31656
| License       | MIT
| Doc PR        | (not needed)

### CODE REVIEW

File: src/Symfony/Component/Intl/Timezones.php [hunk 1]
```diff
@@ -109,6 +109,10 @@ public static function forCountryCode(string $country): array
                 return [];
             }
 
+            if (\strtoupper($country) !== $country) {
+                throw new RuntimeException(sprintf('Country codes must be in uppercase. Did you mean "%s(\'%s\')"?', __METHOD__, \strtoupper($country)));
```
[the remark this answers is not in the document]
Comment (ro0NL) [reply src/Symfony/Component/Intl/Timezones.php]: s/must be/should be/

[the remark this answers is not in the document]
Comment (javiereguiluz) [reply src/Symfony/Component/Intl/Timezones.php]: I'm confused about this:

* "must" = mandatory
* "should" = optional

But uppercase codes are mandatory, right?

[the remark this answers is not in the document]
Comment (ro0NL) [reply src/Symfony/Component/Intl/Timezones.php]: yes, but we dont know if the uppercase variant will exist. The current hint kinda implies changing the case from lower to upper will result in a successful call.. which isnt true per se. 

`forCountryCode('BLA')` will still not work, but we **must** change it as such when calling `forCountryCode('bla')`. I think "should" clarifies something **might** be wrong.

[the remark this answers is not in the document]
Comment (javiereguiluz) [reply src/Symfony/Component/Intl/Timezones.php]: It totally makes sense now. According to your comments, I've changed the logic to display the error message only if the uppercased code exists as a country.

**Review by ro0NL — APPROVED** [review]

**Comment by ro0NL** [comment]
(for 4.3?)

File: src/Symfony/Component/Intl/Timezones.php [hunk 2]
```diff
@@ -109,6 +109,10 @@ public static function forCountryCode(string $country): array
                 return [];
             }
 
+            if (Countries::exists(\strtoupper($country))) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Timezones.php]: please remove the leading `\`

File: src/Symfony/Component/Intl/Timezones.php [hunk 3]
```diff
@@ -109,6 +109,10 @@ public static function forCountryCode(string $country): array
...
+                throw new MissingResourceException(sprintf('Country codes must be in uppercase. Did you mean "%s(\'%s\')"?', __METHOD__, \strtoupper($country)));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Timezones.php]: leading `\` here too

Comment (nicolas-grekas) [reply src/Symfony/Component/Intl/Timezones.php]: what about removing the method and suggesting only the uppercase country code?

**Review by nicolas-grekas — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @javiereguiluz.

**Pull request merged**